queue/queue.go

```diff
@@ -30,6 +30,11 @@ type Queue struct {
 
 type Metrics beatsqueue.Metrics
 
+// EntryID is a unique ascending id assigned to each entry that goes in the
+// queue, to handle acknowledgments within the shipper and report progress
+// to the client.
+type EntryID uint64
```

Comment:
Would not it be better to call it `Index` like we do in the protobuf? I'd like to have consistent naming to avoid confusion.